plugins/catalog-backend-module-bitbucket-server/src/providers/BitbucketServerEntityProvider.ts

```diff
@@ -218,10 +261,251 @@ export class BitbucketServerEntityProvider implements EntityProvider {
+    ) {
+      return true;
+    }
+
+    if (!this.eventConfigErrorThrown) {
+      this.eventConfigErrorThrown = true;
+      throw new Error(
+        `${this.getProviderName()} not well configured to handle repo:push. Missing CatalogApi and/or TokenManager.`,
+      );
+    }
+
+    return false;
+  }
+
+  private async getLocationEntity(
+    event: BitbucketServerEvents.RefsChangedEvent,
+  ): Promise<Entity[]> {
+    const client = BitbucketServerClient.fromConfig({
+      config: this.integration.config,
+    });
+    const result: Entity[] = [];
+    try {
+      const repository = await client.getRepository({
+        projectKey: event.repository.project.key,
+        repo: event.repository.slug,
+      });
+
+      for await (const entity of this.parser({
+        client,
+        logger: this.logger,
+        location: {
+          type: 'url',
+          target: `${repository.links.self[0].href}${this.config.catalogPath}`,
+          presence: 'optional',
+        },
+      })) {
+        entity.metadata.annotations![
+          this.targetAnnotation
+        ] = `${repository.links.self[0].href}${this.config.catalogPath}`;
+
+        if (entity.metadata.annotations === undefined) {
+          entity.metadata.annotations = {};
+        }
+
+        if (repository.defaultBranch === undefined) {
+          const defaultBranchResponse = await client.getDefaultBranch({
+            repo: repository.slug,
+            projectKey: event.repository.project.key,
+          });
+          entity.metadata.annotations[this.defaultBranchAnnotation] =
+            defaultBranchResponse.displayId;
+        } else {
+          entity.metadata.annotations[this.defaultBranchAnnotation] =
+            repository.defaultBranch;
+        }
+        result.push(entity);
+      }
+    } catch (error: any) {
+      if (error.name === 'NotFoundError') {
+        this.logger.error(error.message);
+      }
+    }
+
+    return result;
+  }
+
+  private async onRepoPush(
+    event: BitbucketServerEvents.RefsChangedEvent,
+  ): Promise<void> {
+    if (!this.canHandleEvents()) {
+      this.logger.error(
+        'Bitbucket Server catalog entity provider is not set up to handle events. Missing authService or catalogApi.',
+      );
+      return;
+    }
+
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+
+    const repoSlug = event.repository.slug;
+    const catalogRepoUrl: string = `https://${this.config.host}/projects/${event.repository.project.key}/repos/${repoSlug}/browse${this.config.catalogPath}`;
+    this.logger.info(`handle repo:push event for ${catalogRepoUrl}`);
+    const targets = await this.getLocationEntity(event);
+    if (targets.length === 0) {
+      this.logger.error('Failed to create location entity.');
+      return;
+    }
+    const { token } = await this.auth!.getPluginRequestToken({
+      onBehalfOf: await this.auth!.getOwnServiceCredentials(),
+      targetPluginId: 'catalog', // e.g. 'catalog'
```

Comment:
```suggestion
      targetPluginId: 'catalog',
```

:) But again, if you switch to `CatalogService`, you will pass it `this.auth.getOwnServiceCredentials()` directly instead of making a token at all, so you won't have to give a plugin ID either